modular_skyrat/modules/reagent_forging/code/forge.dm

```diff
@@ -462,6 +467,40 @@
 		COOLDOWN_START(spawned_glass, remaining_heat, glassblowing_amount)
 		return TRUE
 
+	if(istype(attacking_item, /obj/item/food))
+		var/obj/item/food/thing_to_cook = attacking_item
+		if(in_use)
+			to_chat(user, span_warning("You cannot do multiple things at the same time!"))
+			return
+		in_use = TRUE
+		if(forge_temperature < MIN_FORGE_TEMP)
+			fail_message(user, "The [src] is not hot enough to start cooking [thing_to_cook]!")
+			return
+		var/datum/user_input = tgui_input_list(user, "Select a machine to emulate the cooking effects of.", "Cooking Selection", cooking_choice)
```

Comment:
Honestly, I think a radial menu would be nicer here. See how the reagent grinder does it. Would need some new icons, but they can just be smaller versions of the oven and microwave respectively.
Something like;
```suggestion
	    var/static/radial_oven = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_oven")
	    var/static/radial_microwave = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_microwave")
        var/list/options = list()
        options["oven"] = radial_oven
        options["microwave"] = radial_microwave
        if(length(options) < 1)
		    return
	    if(length(options) == 1)
		    for(var/key in options)
			    choice = key
	    else
		    choice = show_radial_menu(user, src, options)
```

Reply:
I had thought of making it a radial menu, just didn't make one out of not knowing how they work, let me see what I can do